kubectl-fdb/cmd/cordon_test.go

```diff
@@ -21,121 +21,219 @@
-							fdbv1beta2.FDBClusterLabel:        clusterName,
-							fdbv1beta2.FDBProcessGroupIDLabel: "instance-2",
-						},
-					},
-					Spec: corev1.PodSpec{
-						NodeName: "node-2",
-					},
-				},
-			}
+			// creating Pods for first cluster.
+			Expect(createPods(clusterName, namespace)).NotTo(HaveOccurred())
 
-			for _, pod := range pods {
-				Expect(k8sClient.Create(context.TODO(), &pod)).NotTo(HaveOccurred())
-			}
+			// creating a second cluster
+			secondCluster = generateClusterStruct(secondClusterName, namespace)
+			Expect(k8sClient.Create(context.TODO(), secondCluster)).NotTo(HaveOccurred())
+			Expect(createPods(secondClusterName, namespace)).NotTo(HaveOccurred())
 		})
 
 		DescribeTable("should cordon all targeted processes",
 			func(input testCase) {
-				err := cordonNode(k8sClient, cluster, input.nodes, namespace, input.WithExclusion, false, 0)
+				// We use these buffers to check the input/output
+				outBuffer := bytes.Buffer{}
+				errBuffer := bytes.Buffer{}
+				inBuffer := bytes.Buffer{}
+
+				cmd := newAnalyzeCmd(genericclioptions.IOStreams{In: &inBuffer, Out: &outBuffer, ErrOut: &errBuffer})
+				err := cordonNode(cmd, k8sClient, input.clusterName, input.nodes, namespace, input.WithExclusion, false, 0, input.clusterLabel)
 				Expect(err).NotTo(HaveOccurred())
 
-				var resCluster fdbv1beta2.FoundationDBCluster
-				err = k8sClient.Get(context.Background(), client.ObjectKey{
-					Namespace: namespace,
-					Name:      clusterName,
-				}, &resCluster)
+				clusterNames := []string{clusterName, secondClusterName}
+				var instancesToRemove []string
+				var instancesToRemoveWithoutExclusion []string
+				for _, clusterName := range clusterNames {
+					var resCluster fdbv1beta2.FoundationDBCluster
+					err = k8sClient.Get(context.Background(), client.ObjectKey{
+						Namespace: namespace,
+						Name:      clusterName,
+					}, &resCluster)
+					instancesToRemove = append(instancesToRemove, resCluster.Spec.ProcessGroupsToRemove...)
+					instancesToRemoveWithoutExclusion = append(instancesToRemoveWithoutExclusion, resCluster.Spec.ProcessGroupsToRemoveWithoutExclusion...)
+				}
 
 				Expect(err).NotTo(HaveOccurred())
 				// Use equality.Semantic.DeepEqual here since the Equal check of gomega is to strict
-				Expect(equality.Semantic.DeepEqual(input.ExpectedInstancesToRemove, resCluster.Spec.ProcessGroupsToRemove)).To(BeTrue())
-				Expect(equality.Semantic.DeepEqual(input.ExpectedInstancesToRemoveWithoutExclusion, resCluster.Spec.ProcessGroupsToRemoveWithoutExclusion)).To(BeTrue())
+				fmt.Printf("\ninstancesToRemove: %v\n", instancesToRemove)
+				Expect(equality.Semantic.DeepEqual(input.ExpectedInstancesToRemove, instancesToRemove)).To(BeTrue())
+				fmt.Printf("\ninstancesToRemoveWithoutExclusion: %v\n", instancesToRemoveWithoutExclusion)
+				Expect(equality.Semantic.DeepEqual(input.ExpectedInstancesToRemoveWithoutExclusion, instancesToRemoveWithoutExclusion)).To(BeTrue())
```

Comment:
```suggestion
				Expect(input.ExpectedInstancesToRemove).To(ConsistOf(instancesToRemove))
				Expect(input.ExpectedInstancesToRemoveWithoutExclusion).To(ConsistOf(instancesToRemoveWithoutExclusion))
```

See https://pkg.go.dev/github.com/onsi/gomega#ConsistOf, that check was "recently" added in gomega and provides more details on what element is missing.